providers/azure.go

```diff
@@ -103,9 +108,38 @@ func overrideTenantURL(current, defaultURL *url.URL, tenant, path string) {
 	}
 }
 
+func getMicrosoftGraphGroupsURL(groupField string) *url.URL {
+
+	selectStatement := "$select=displayName,id"
+	if !slices.Contains([]string{"displayName", "id"}, groupField) {
+		selectStatement += "," + groupField
+	}
+
+	// Select only security groups. Due to the filter option, count param is mandatory even if unused otherwise
+	return &url.URL{
+		Scheme:   "https",
+		Host:     "graph.microsoft.com",
+		Path:     "/v1.0/me/transitiveMemberOf",
+		RawQuery: "$count=true&$filter=securityEnabled+eq+true&" + selectStatement,
+	}
+}
+
 func (p *AzureProvider) GetLoginURL(redirectURI, state, _ string, extraParams url.Values) string {
+	// In azure oauth v2 there is no resource param
+	// https://docs.microsoft.com/en-us/azure/active-directory/azuread-dev/azure-ad-endpoint-comparison#scopes-not-resources
+	if p.isV2Endpoint {
+		// In azure oauth v2 there is no groups scope so replace it if exists
+		p.Scope = strings.ReplaceAll(p.Scope, " groups", "")
```

Comment:
Do we really want to modify the users config at runtime or show a warning at boot?

Reply:
`groups` scope is not necessary directly added by the user, it might be computed while configuring the provider data, like for example in https://github.com/oauth2-proxy/oauth2-proxy/blob/master/providers/providers.go#L159. So instead of logging a warning that might confuse the user even more, I prefer to silently remove it as it is not supported in V2 anyway. But I'm open to suggestion, if you consider it makes more sense to add a warning I can add it.

Reply:
@adriananeci I would like the PR closed as soon as possible but I'm afraid that giving a warning makes sense.
Maybe in the log, we can explicitly write the reason for the warning.

Reply:
Or we could close this PR and do another one to add the warning so we close this one. 🚀